### ISSUE TITLE
Toolkit is disregarding packageName in favor of subdirectory

### ISSUE THREAD

**Issue by crstauf** [raycast/script-commands#121]
It seems that the toolkit looks at only the subdirectory name and completely disregards the `packageName` meta value. This is demonstrated on [the emoji scripts](https://github.com/raycast/script-commands/tree/master/commands#emojis), which [have the `packageName` set as `Communication`](https://github.com/raycast/script-commands/blob/master/commands/communication/emojis/emojis-search.sh#L12), yet are classified as `Emoji` in the auto-generated documentation. This differs from Raycast behavior (emoji scripts shown as "Communication), and I believe is a bug or oversight.

**Comment by crstauf** [comment]
This would also help solve capitalization issues, such as with `Tv` rather than `TV`.

**Comment by unnamedd** [comment]
> It seems that the toolkit looks at only the subdirectory name and completely disregards the `packageName` meta value. This is demonstrated on [the emoji scripts](https://github.com/raycast/script-commands/tree/master/commands#emojis), which [have the `packageName` set as `Communication`](https://github.com/raycast/script-commands/blob/master/commands/communication/emojis/emojis-search.sh#L12), yet are classified as `Emoji` in the auto-generated documentation. This differs from Raycast behavior (emoji scripts shown as "Communication), and I believe is a bug or oversight.

This is not a bug, this is intentional. Before, when we didn’t have support to nested directories, the use of `packageName` made sense to create a fake subcategory, this is not needed anymore, we use the directory name to create the subcategories in the documents. The use of `packageName` to create a subcategory in the document would only bring more confusion and complexity and we decided to avoid that.

**Comment by unnamedd** [comment]
> This would also help solve capitalization issues, such as with `Tv` rather than `TV`.

This will be solved soon.

**Comment by crstauf** [comment]
> The use of `packageName` to create a subcategory in the document would only bring more confusion and complexity

Raycast itself uses `packageName` for identifying scripts, so how would that create more confusion? Plus, the `packageName` can always match the subdirectory name, now that `packageName` is a required piece of meta data.

**Comment by unnamedd** [comment]
> > The use of `packageName` to create a subcategory in the document would only bring more confusion and complexity
> 
> 
> 
> Raycast itself uses `packageName` for identifying scripts, so how would that create more confusion? Plus, the `packageName` can always match the subdirectory name, now that `packageName` is a required piece of meta data.

First, `packageName` is *not* a required meta data, it's being configured for the Script Commands, but if you remove, the Script Command will continue to be seen by Raycast and working, but try to remove `title`, `schemaVersion` and `mode`, then the Script Command will not work anymore.

Second, the Toolkit isn't Raycast. The Toolkit is a separated and open source tool to generate documentation related to script commands submitted here. 

You mentioned an example of script command above which is wrong. It's placed inside of the folder `emoji`, but you had set the `packageName` as _Communication_, which is in that context wrong. My suggestion for you is to change the `packageName` from _Communication_ to _Emoji_ to keep the consistency with the folder where it is living now.

Third, the confusion might be created when the script command do not respect the subcategory where it's being placed. We have a list of Communications, which is a folder and a category, that contains a script command that is being presented in he category but the file isn't in the Communication folder. Doesn't make any sense.

Fourth, you mentioned couple of times portability, you asked to set the `packageName` because in case of changing the script command from one folder to another, you wouldn't lose the category where it's belong and now, you want to change that.

Sorry, but for now, I suggest you to change the `packageName` to use the right information which should be presented in Raycast, because in the documentation, there's nothing to do for now. The Toolkit is working as planned and the decision done here was in common agreement between me and Thomas and we won't use for now the `packageName` to create subcategories like we did in the past.

**Comment by crstauf** [comment]
{ sigh } Due to the hostility expressed in your comments, @unnamedd, I'm responding in same tone on occasion. 

Here we go:

> First, `packageName` is _not_ a required meta data

Yes, yes it is. See https://github.com/raycast/script-commands/issues/47#issuecomment-726723169 and [_Metadata convention_ section of contributing guidelines](https://github.com/raycast/script-commands/blob/master/CONTRIBUTING.md#metadata-convention).

> Second, the Toolkit isn't Raycast.

This is true, but it is closely related, and based on conversations elsewhere, it is important for this repository to match very closely with Raycast guidelines. 

> You mentioned an example of script command above which is wrong. My suggestion for you is to change the `packageName` from _Communication_ to _Emoji_ to keep the consistency with the folder where it is living now.

It is not wrong. My choice to set `packageName` as `Communication` is intentional. The emoji scripts are placed inside a subdirectory to make turning on/off easy, but the `packageName` should be `Communication`. That the Toolkit does not respect this is improper (hence this issue). Consider for example the IP address scripts (they're in the `ip` subdirectory, but have a `packageName` of `Internet`): `IP` makes little sense as a category name, whereas `Internet` makes more sense. A `packageName` of `IP Address` would make even _more_ sense, but with the Toolkit in its current state, it doesn't matter.

> Third, the confusion might be created when the script command do not respect the subcategory where it's being placed.

As I understand it, the purpose of subdirectories is to make it easy for users to turn scripts on and off within Raycast, not for documentation purposes. There are cases where having scripts in a subdirectory and a top-level `packageName` makes sense (hence this issue, and see previous example).

> Fourth, you mentioned couple of times portability, you asked to set the `packageName` because in case of changing the script command from one folder to another, you wouldn't lose the category where it's belong and now, you want to change that.

I do not want to change it: `packageName` should be and is required (see first point). The distinction I'm making that you seem to be lost on is that `packageName` will not and does not need to be identical to the subdirectory it is within.

> I suggest you to change the `packageName` to use the right information

"The _right_ information"? That's arrogant. As mentioned previously, `packageName` does not serve the same purpose as the subdirectory.

> because in the documentation, there's nothing to do for now.

Hence this issue, to present a reason for it to be changed.

> common agreement between me and Thomas

Interested in the conversation you and @thomaspaulmann had in private becoming public... ya know, in an open source and community kinda way.

>  and we won't use for now the `packageName`

The "for now" is key... this issue seeks to discuss and change so "for now" is "in the past".

It is unfortunate that a civil and "open source" conversation was not allowed to occur about this. Fingers crossed that Raycast developers can enforce a positive community discussion.

**Comment by crstauf** [comment]
> > This would also help solve capitalization issues, such as with `Tv` rather than `TV`.
> 
> This will be solved soon.

It is possible to anticipate every single phrase with abnormal capitalization? Looking forward to seeing that.

**Comment by PitNikola** [comment]
Hey folks, is there anything actionable from this discussion? If not, I will close the issue then.

As to the Toolkit and `packageName` topic, we have no strong opinion here and leaving this decision to @unnamedd. This nicely formatted directory was not something we were planning to do and is basically a gift from Thiago to Raycast and the community, for which we are very thankful.

Personally, I think either of structures could work (folder based and packageName based). 
I also would like to clarify something I mentioned before in another issue about `packageName`: for Raycast this field is nothing but a subtitle right now which can be used as a category or extension name. There is no logic inside Raycast that would bundle scripts under it or use it as identifier. The requirement to provide it in this repository was introduced only to improve portability (e.g. when a person moves script to `_enabled-commands` directory).

I also do hope we can keep this conversation in a civil way, thanks for everyone's understanding.

**Comment by crstauf** [comment]
My proposal is for the Toolkit to use `packageName` if present, and default to the subdirectory if not. The use of `packageName` allows for a script to be independent from the subdirectory it is in, but also because `packageName` is required, could also be the subdirectory's name.

The clearest example of this is the emoji scripts: they're in the `communciation/emoji` directory, but `packageName` is set to `Communication`. Having a separate category for these scripts doesn't make sense, and so I specified `Communication` as the `packageName`, but it was requested that they be put in their own subdirectory to make it easier for users to de/activate.

Another example would be GitHub. There are GitHub scripts both in `developer-utils` and `web-searches`... if the Toolkit respected `packageName`, then there would be a single category in the documentation for GitHub scripts. If the script contributor wanted to include the script in the subdirectory, then they set that as the `packageName`.

`packageName` provides superior flexibility over subdirectory name, and overcomes challenges (such as the capitalization issue mentioned previously).

**Comment by unnamedd** [comment]
Hey @PitNikola, 

there are two points that are good to be mentioned here. The first is related to the use of the `packageName`, which was used by the Toolkit before to create "fake subcategories", we had folders where we read the script metadata from each file, without have to be concerned to which folder the script had belong. If we didn't have the `packageName` configured, I've had just to append that script command under the main category, if we had, then we created a sub category based on the script commands created with that `packageName`. As soon the support for the nested directories was introduced, then we started to use only the folder as the main category/sub category, instead of create a fake sub category.

The second point is, I agree that the use of the `packageName` is something great and I miss it too, because it solves some problemas that I don't have to solve manually, but in the other hand creates a complexity that I don't want to have for now. Create automation is amazing but can also bring headaches and in my personal backlog for features I want to implement for Toolkit is a bit full for while. 

Maybe soon, I will have time to reconsider the use of the recursivity together with the `packageName` metatag, but there are more important things on top of my personal list which I'm going to work before work in this topic again. I'm pretty sure this topic is far from important for the moment.

And to answer your question Petr, I don't think we have anything actionable for while in this topic, because of the points I mentioned earlier. 

I am always open to discuss any topic related to Toolkit and I will always be. I'm really excited to implement more useful features in Toolkit for the whole community. The Toolkit itself is in the very beginning of it's life. The request made in this issue, definitely is not something with a ultra high priority at the moment, it is not a bug and respect the folder structure we have in the repo, that being said, it can wait.
